lightwood/api/data_source.py

```diff
@@ -55,7 +56,7 @@ def __setattr__(self, name, value):
 
 
 class DataSource(Dataset):
-    def __init__(self, data_frame, config, prepare_encoders=True):
+    def __init__(self, data_frame, config, prepare_encoders=True, initialize_transformer=True):
```

Comment:
Is the addition of `initialize_transformer` purely a measure for optimization ? I.e. so that no new ones get created when we create a SubSet.

Or is it somehow necessary for this PR ? If not, I think it's a very good catch that should have been spotted a while ago, but please consider making separate PRs for these in the future.

Reply:
it's used for creating instances of `DataSource` in methods `DataSource.make_child`, `DataSource.subset`